pkg/apis/rke.cattle.io/v1/plan/plan.go

```diff
@@ -74,9 +74,10 @@ type File struct {
 
 // NodePlan is the struct used to deliver instructions/files/probes to the system-agent, and retrieve feedback
 type NodePlan struct {
-	Files                []File                `json:"files,omitempty"`
-	Instructions         []OneTimeInstruction  `json:"instructions,omitempty"`
-	PeriodicInstructions []PeriodicInstruction `json:"periodicInstructions,omitempty"`
-	Error                string                `json:"error,omitempty"`
-	Probes               map[string]Probe      `json:"probes,omitempty"`
+	Files                                 []File                `json:"files,omitempty"`
+	Instructions                          []OneTimeInstruction  `json:"instructions,omitempty"`
+	PeriodicInstructions                  []PeriodicInstruction `json:"periodicInstructions,omitempty"`
+	Error                                 string                `json:"error,omitempty"`
+	Probes                                map[string]Probe      `json:"probes,omitempty"`
+	ResetFailureCountOnSystemAgentRestart bool                  `json:"resetFailureCountOnStartup,omitempty"`
```

Comment:
I'm a fan of having thename and json tag match, WDYT?

Reply:
I am too in general, but in this case I think the difference makes sense. The json tag has to match in both the system-agent repo and rancher in order for the plan to be handed off via the secret. It doesn't make sense to have the json tag include the term `system-agent` inside the system-agent repo.